### DISCUSSION TITLE
I published a plugin on npm, how long does it take to show in BABEL REPL?

### DISCUSSION THREAD

**Discussion by diraneyya** [babel/babel#15458] [General] [answered]
Hello lovelies 💞

I forked an inline-function preprocessor from GitHub user @chocolateboy and published it under npm with different defaults (basically inlining all functions).

The published npm package can be found [here](https://www.npmjs.com/package/babel-plugin-inline-all-functions)

I am now wondering what would be the remaining steps to make it appear in the BABEL REPL online tool, so that I can start using it in my classes (I teach Javascript for a living).

Orwa.

**Comment by JLHwung** [comment, marked as answer]
It may take some time before layers of cache service catch up with the new plugin. If you are in a hurry, you can try

https://astexplorer.net/#/gist/41518929454821acb57ead585f9c8897/c6897752d07fc21a0b1c9e0df37ded3d9fd549b4

where you can modify the plugin on the fly and check out the transform result / AST.

**Reply by diraneyya** [reply]
I really cannot express enough gratitude for this post. I literally can develop a BABEL transformers in babysteps and `console.dir` anything to the console while seeing everything, the source code, the destination code, the tree, and edit my transformer code all in one place and on one screen, inside of a browser instance! It is a remarkable approach to learning how to do this stuff. Thank you for this! 😍😍😍

**Reply by JLHwung** [reply]
Glad that it helps! Enjoy hacking.

**Reply by diraneyya** [reply]
For some reason I am still unable to use it in the BABEL REPL. Any idea why this might be the case?

This is my plugin: https://www.npmjs.com/package/babel-plugin-inline-all-functions?activeTab=readme

<img width="499" alt="image" src="https://user-images.githubusercontent.com/107997002/221386589-0884785a-adaa-4c2d-9e6f-aea317696516.png">
<img width="596" alt="image" src="https://user-images.githubusercontent.com/107997002/221386608-54983c93-a446-4e0e-8119-27b05ce45371.png">

**Comment by diraneyya** [comment]
Incredible

<img width="1412" alt="image" src="https://user-images.githubusercontent.com/107997002/221355340-31512af0-1f30-48e5-9209-13ad93bb578c.png">
